warehouse/query-core/src/main/java/datawave/query/function/LimitFields.java

```diff
@@ -232,57 +274,153 @@ private boolean isHit(String keyWithGrouping, Attribute<?> attr, Multimap<String
+        String[] splits = StringUtils.split(key, '.');
         if (splits.length >= 3) {
-            // return the first group and last group (a.k.a the instance in the first group)
-            return new String[] {splits[1], splits[splits.length - 1]};
+            // Return the first group and last group (aka the instance in the first group).
+            return Optional.of(Pair.with(splits[1], splits[splits.length - 1]));
         }
-        return null;
+        return Optional.empty();
     }
     
+    /**
+     * Return the hit term map for the given document.
+     * 
+     * @param document
+     *            the document
+     * @return the hit term map
+     */
     private Multimap<String,String> getHitTermMap(Document document) {
         Multimap<String,String> attrMap = HashMultimap.create();
         fillHitTermMap(document.get(JexlEvaluation.HIT_TERM_FIELD), attrMap);
         return attrMap;
     }
     
-    private void fillHitTermMap(Attribute<?> attr, Multimap<String,String> attrMap) {
-        if (attr != null) {
-            if (attr instanceof Attributes) {
-                Attributes attrs = (Attributes) attr;
-                for (Attribute<?> at : attrs.getAttributes()) {
-                    fillHitTermMap(at, attrMap);
-                }
-            } else if (attr instanceof Content) {
-                Content content = (Content) attr;
-                // split the content into its fieldname:value
-                String contentString = content.getContent();
-                attrMap.put(contentString.substring(0, contentString.indexOf(":")), contentString.substring(contentString.indexOf(":") + 1));
+    /**
+     * Add each hit term and value found in the given attribute to the given map.
+     * 
+     * @param attribute
+     *            the attribute
+     * @param map
+     *            the map
+     */
+    private void fillHitTermMap(Attribute<?> attribute, Multimap<String,String> map) {
+        if (attribute != null) {
+            if (attribute instanceof Attributes) {
+                Attributes attributes = (Attributes) attribute;
+                attributes.getAttributes().forEach((child) -> fillHitTermMap(child, map));
+            } else if (attribute instanceof Content) {
+                // Split the content into its fieldName:value and add it to the map.
+                String contentString = ((Content) attribute).getContent();
+                int delimiterPos = contentString.indexOf(":");
+                String fieldName = contentString.substring(0, delimiterPos);
+                String value = contentString.substring(delimiterPos + 1);
+                map.put(fieldName, value);
             }
         }
     }
     
-    protected String removeGrouping(String key) {
-        // if we have grouping context on, remove the grouping context
+    /**
+     * If the key begins with grouping context, return a subset of the key without the grouping context.
```

Comment:
"begins" is a little ambiguous here.  It removes the grouping context from the key if present.

Reply:
I edited the documentation to remove the ambiguous parts.